tensorflow/python/framework/python_op_gen.cc

```diff
@@ -355,18 +416,97 @@ string GenEagerPythonOp::Code() {
   string eager_not_allowed_error = GetEagerNotAllowedError();
 
   if (!AddEagerFastPathAndGraphCode(parameters_with_defaults, output_sizes,
-                                    eager_not_allowed_error)) {
+                                    eager_not_allowed_error, type_annotations)) {
     return result_;
   }
 
   if (!AddEagerFallbackCode(parameters, output_sizes, num_outputs_expr,
-                            eager_not_allowed_error)) {
+                            eager_not_allowed_error, type_annotations)) {
     return result_;
   }
 
   return prelude_ + result_;
 }
 
+std::unordered_map<string, string> GenEagerPythonOp::GetTypeAnnotationMap() {
+  std::unordered_map<string, string> type_annotations;
+  // Mapping attrs to TypeVars
+  for (const auto& attr : op_def_.attr()) {
+    if (attr.type() == "type") {
+      const string type_var_name = "TV_" + op_def_.name() + "_" + attr.name();
+      type_annotations[attr.name()] = type_var_name;
+    } else if (attr.type() == "bool" || attr.type() == "float" ||
+               attr.type() == "int" || attr.type() == "bytes") {
+      type_annotations[attr.name()] = attr.type();
+    }
+  }
+
+  // Mapping input Tensors to their types
+  for (const auto& arg : op_def_.input_arg()) {
+    // Do not add type annotations to args that accept a sequence of Tensors
```

Comment:
Let's make this a TODO (and it would make a nice addition in a separate PR, too).